services/core/shared/core/delegates.js

```diff
@@ -76,16 +87,22 @@ const getDelegates = async params => {
 			const dbResult = await db.find(inputData);
 
 			if (dbResult.length > 0) delegates.data = dbResult;
-			else throw new Error('Request data from Lisk Core');
+			else throw new Error('Request delegates data from Lisk Core');
 		}
 	} catch (err) {
 		logger.debug(err.message);
 
 		delegates = await coreApi.getDelegates(params);
-		delegates.data.map(delegate => {
-			delegate.id = String(delegate.rank);
-			return delegate;
-		});
+		delegates.data = await BluebirdPromise.map(
+			delegates.data,
+			async delegate => {
+				const dbResult = await db.find(getSelector({ address: delegate.account.address }));
+				if (dbResult.length) [delegate] = dbResult;
+				else delegate.id = String(delegate.rank);
+				return delegate;
+			},
+			{ concurrency: delegates.data.length },
```

Comment:
Since delegates are stored in memory, logic that retrieves them from the server can be out of the getDelegates() function. Use `requestAll.js` to retrieve all delegates and keep them in memory. `getDelegates()` can retrieve data from the delegate store only.

Reply:
Done!